backend/models/office_hours/oh_event.py

```diff
@@ -0,0 +1,36 @@
+from pydantic import BaseModel
+from datetime import datetime, date
+from oh_type import OfficeHoursType
+
+__authors__ = ["Sadie Amato, Bailey DeSouza, Meghan Sun, Maddy Andrews"]
+__copyright__ = "Copyright 2024"
+__license__ = "MIT"
+
+
+class OfficeHoursEventDraft(BaseModel):
+    """
+    Pydantic model to represent an `OfficeHoursEvent` that has not been created yet.
+
+    This model is based on the `OfficeHoursEventEntity` model, which defines the shape
+    of the `OfficeHoursEvent` database in the PostgreSQL database
+    """
+
+    office_hours_section_id: int
+    room_id: int
```

Comment:
I would suggest raising the level of abstraction in the models here, but doing so may mean tweaking the models of `OfficeHoursSection` and `Room` slightly.

The suggested change would be adding a `RoomPartial` model to `backend/models/room.py` and making `nickname` be `| None`: 

~~~
class RoomPartial(Room):
   nickname: str | None = None
~~~ 

Similarly, with `OfficeHoursSection`:

~~~
class OfficeHoursSectionPartial(OfficeHours):
  title: str | None = None
~~~

The reason for moving toward partials (an idea inspired by TypeScript which has a true notion of a Partial version of a type where fields can be null) is so the client can pass the full object if it'd like and not have to reshape the request to form a `OfficeHoursEventDraft` object to extract `id`s of foreign key relationships (which is hopefully a lower level abstraction, even though it winds up getting passed in anyway). 

This way, from the client-side, you'll be able to make requests like:

~~~
this.httpClient.post('/office-hours/event', {
   office_hours_section: officeHoursSection,
   room: room,
   type: type,
   /* ... */
});
~~~

Notice the slight nicety that on the client side you can think in terms of the objects you otherwise have in the system (like the officeHoursSection object and room object) rather than having to remember to pull out their IDs each time.

The primary motivation here is trying to give the client the ability to think less about the database structure (where do I have to pull out an `id` in my request vs. just give an object) and more about the objects in the system.

Finally, this helps avoid accidental errors such as choosing the wrong type for an `id`. The Room type's `id` is actually a `str`!

Reply:
This has been adjusted now, thank you!!